mass_mailing_switzerland/models/res_partner.py

```diff
@@ -165,8 +165,15 @@ def write(self, vals):
                     break
         super().write(vals)
         if update_partner_ids and not self.env.context.get("import_from_mailchimp"):
-            self.env["mail.mass_mailing.contact"]\
-                .with_delay().update_all_merge_fields_job(update_partner_ids)
+            for partner_id in update_partner_ids:
+                queue_job = self.env["queue.job"].search([
+                    ("channel", "=", "root.mass_mailing_switzerland.update_partner_mailchimp"),
+                    ("state", "!=", "done")
+                ])
+                if len(queue_job) and partner_id in queue_job.mapped("args"):
+                    continue
+                self.env["mail.mass_mailing.contact"] \
+                    .with_delay().update_partner_merge_fields_job(partner_id)
```

Comment:
I think this logic could be moved inside the job function `update_partner_merge_fields_job` so that here we wouldn't need to change the code (plus this would be done each time the job is called instead of just here).

Reply:
:+1: 
I will use `update_all_merge_fields_job` since it can be call with a list of partners and will launch a job for each one of them. It also take care of the delay call